python/dataproc_templates/pubsublite/README.md

```diff
@@ -0,0 +1,73 @@
+Template for reading files from Cloud Pub/sub Lite and writing them to a BigQuery table.
+
+It uses the [Pubsublite Spark SQL Streaming](https://central.sonatype.com/artifact/com.google.cloud/pubsublite-spark-sql-streaming/1.0.0) to read the streaming data from Pubsublite and [Spark BigQuery connector](https://cloud.google.com/dataproc-serverless/docs/guides/bigquery-connector-spark-example) for writing to BigQuery
+
+## Arguments
+
+* `pubsublite.to.bq.input.subscription`: Cloud Storage location of the input files (format: `gs://bucket/...`)
+* `pubsublite.to.bq.write.mode`: Output write mode (one of: append,overwrite,ignore,errorifexists)(Defaults to append)
+* `pubsublite.to.bq.output.dataset`: BigQuery output dataset name
+* `pubsublite.to.bq.output.table`: BigQuery output table name
+* `pubsublite.to.bq.bucket.name`: Temporary bucket for the Spark BigQuery connector
+
+## Usage
+
+```
+$ python main.py --template PUBSUBLITETOBQ --help
+usage: main.py --template PUBSUBTOBQ [-h] \
+    --pubsublite.to.bq.input.subscription PUBSUBLITE.BIGQUERY.INPUT.SUBSCRIPTION \
+    --pubsublite.to.bq.project.id PUBSUBLITE.BIGQUERY.OUTPUT.PROJECT.ID \
+    --pubsublite.to.bq.output.dataset PUBSUBLITE.BIGQUERY.OUTPUT.DATASET \
+    --pubsublite.to.bq.output.table PUBSUBLITE.BIGQUERY.OUTPUT.TABLE \
+    --pubsublite.to.bq.bucket.name PUBSUBLITE.BIGQUERY.BUCKET.NAME \
+    [--pubsublite.to.bq.write.mode {overwrite,append,ignore,errorifexists}] \
+    --pubsublite.to.bq.checkpoint.location PUBSUBLITE.BIGQUERY.CHECKPOINT.LOCATION
```

Comment:
Is this supported? if yes, please test the functionality as it was not tested in spike task.

Reply:
`**_'pubsublite.to.bq.**_checkpoint.location_**': 'gs://arindamsarkar-test/pubsubtogcs-**_checkpoint-location_**',_**`
 'pubsublite.to.bq.input.subscription': 'arindamsarkar-test-sub-lite',
 'pubsublite.to.bq.input.timeout.ms': 60000,
 'pubsublite.to.bq.input.topic': None,
 'pubsublite.to.bq.output.dataset': 'arindamsarkar_test',
 'pubsublite.to.bq.output.table': 'pubsubtobq',
 'pubsublite.to.bq.project.id': 'yadavaja-sandbox',
 'pubsublite.to.bq.streaming.duration.seconds': 15,
 'pubsublite.to.bq.total.receivers': 5,
 'pubsublite.to.bq.write.mode': 'overwrite'}
23/03/06 10:17:28 INFO StateStoreCoordinatorRef: Registered StateStoreCoordinator endpoint
`23/03/06 10:17:28 INFO ResolveWriteToStream: **_Checkpoint root gs://arindamsarkar-test/pubsubtogcs-checkpoint-location resolved to gs://arindamsarkar-test/pubsubtogcs-checkpoint-location._**`
23/03/06 10:17:28 WARN ResolveWriteToStream: spark.sql.adaptive.enabled is not supported in streaming DataFrames/Datasets and will be disabled.
`23/03/06 10:17:29 INFO MicroBatchExecution: Starting [id = 4c8219ed-abc7-4768-9cb9-1ae0e40154dc, runId = **_9d059bfd-6870-47aa-bbf0-4cb578006b52]. Use gs://arindamsarkar-test/pubsubtogcs-checkpoint-location to store the query checkpoint._**`

Highlighted some of the places in the logs where the checkpoint location is being used while executing the template. So, based on these it seems it should be supported.